tools/src/keygen/KeygenSecret.hs

```diff
@@ -45,7 +45,7 @@ parseInputWithSecret = \case
         content <- decode $ LBS.fromStrict input
         fromKeyfileContent pp content
     CommSecret n -> \input ->
-        pure $ committeeDerive (CommitteeSecret input) n
+        pure $ committeeDerive (leftToPanic $ mkCommitteeSecret input) n
```

Comment:
why `leftToPanic`? this is a method which returns `Maybe` exactly to be able to fail gracefully, you should use `rightToMaybe` here.